### DISCUSSION TITLE
UnmarshalText not working?

### DISCUSSION THREAD

**Discussion by fugkco** [pelletier/go-toml#908] [Questions]
I don't think this is a bug, I've a feeling I'm just being a dunce. But I'm struggling to get the code to go into `TextUnmarshaler` interface. Anyone know what I'm doing wrong?

```
package main

import (
	"errors"
	"testing"

	"github.com/pelletier/go-toml/v2"
)

type Packages map[string]any

type Config struct {
	Packages Packages `toml:"package"`
}

func (b *Packages) UnmarshalText(text []byte) error {
	return errors.New("never reached")
}

func TestUnmarshal(t *testing.T) {
	doc := []byte(`
[package."mypkg"]
somekey = "somevalue"
`)

	v := Config{}

	err := toml.Unmarshal(doc, &v)
	if err == nil {
		t.Fail()
	}
}
```

**Comment by fugkco** [comment]
ping @pelletier

**Comment by pelletier** [comment]
Sorry I missed this question; thank you for the ping.

The `UnmarshalText` method in this example is never called because this interface is only used on TOML strings. In this case, the interface is implemented on the table (modeled as a map in this example).

See the [TextUnmarshal](https://pkg.go.dev/github.com/pelletier/go-toml/v2#example-Unmarshal-TextUnmarshal) example. Also maybe this issue of somebody having a similar problem: https://github.com/pelletier/go-toml/issues/865.

**Reply by fugkco** [reply]
Is there any way to unmarshal the whole table instead of individual primitive types?

**Reply by pelletier** [reply]
At the moment the two potential features you may be interested in are todos (support `toml.RawMessage` https://github.com/pelletier/go-toml/issues/796 and `toml.Unmarshaler` https://github.com/pelletier/go-toml/issues/873).

If you want to describe your use-case maybe I can help find a workaround in the meantime!

**Reply by fugkco** [reply]
I guess I'm just trying to unmarshal a value that can be different objects. I'll see if I can find a different method to unmarshal it.
